cmd/node/node.go

```diff
@@ -360,6 +363,45 @@ func (app SpacemeshApp) stopServices() {
 
 }
 
+func isFileExist(name string) (bool, error) {
+	_, err := os.Stat(name)
+	if os.IsNotExist(err) {
+		return false, nil
+	}
+	return err != nil, err
+}
+
+func getEdIdentity() (*signing.EdSigner, error) {
+	dataDir, err := filesystem.GetSpacemeshDataDirectoryPath()
+	if err != nil {
+		log.Error("Could not get data path err=%v", err)
+		return signing.NewEdSigner(), err
+	}
+
+	f := dataDir + "/identity.ed"
+	if exist, _ := isFileExist(f); exist {
+		log.Warning("Identity file not found. Creating new identity")
+		edSgn := signing.NewEdSigner()
+		ioutil.WriteFile("identity.ed", edSgn.ToBuffer(), 0644)
+		return edSgn, nil
+	}
+
+	log.Info("Identity file exist. Reading identity from file")
+	buff, err := ioutil.ReadFile(f)
+	if err != nil {
+		log.Error("Could not read identity from file err=%v", err)
+		return nil, err
+	}
+
```

Comment:
Add a sanity check to test that the content of the file is correct (check that the private and public key match)

Reply:
I added a sign-verify sanity to the construction from buffer